authservice/auth_service.go

```diff
@@ -325,47 +415,110 @@ func (r *RemoteAuthDataLoaderImpl) GetAccessToken() error {
 		return fmt.Errorf("error getting access token: %d - %s", resp.StatusCode, string(body))
 	}
 
-	err = json.Unmarshal(body, &r.accessToken)
+	var accessToken AccessToken
+	err = json.Unmarshal(body, &accessToken)
 	if err != nil {
 		return fmt.Errorf("error on unmarshal access token response: %v", err)
 	}
 
+	r.accessTokens.Store(fmt.Sprintf("%s_%s", appID, orgID), accessToken)
+
 	return nil
 }
 
+func (r *RemoteAuthDataLoaderImpl) isAppOrgAccessGranted(appID string, orgID string) bool {
+	granted := false
+	for _, pair := range r.appOrgPairs {
+		if pair.AppID == appID && pair.OrgID == orgID {
+			granted = true
+			break
+		}
+	}
+
+	return granted
+}
+
+//GetAccessTokens returns a map containing all cached access tokens
+func (r *RemoteAuthDataLoaderImpl) GetAccessTokens() map[string]string {
```

Comment:
As discussed, it would probably be better to allow all service account tokens (all app/org pair tokens) to be retrieved at once. This will improve efficiency, and make this process simpler. I have opened https://github.com/rokwire/core-building-block/issues/459 to add the necessary API to the Core BB.